src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpCodeVisitor.cs

```diff
@@ -515,5 +515,28 @@ public static CSharpCodeWriter RenderPreWriteStart(CSharpCodeWriter writer, Code
 
             return writer;
         }
+
+        /// <summary>
+        /// Renders the Write method for TagHelpers.
+        /// </summary>
+        /// <param name="writer"><see cref="CSharpCodeWriter"/> to which the output must be written.</param>
+        /// <param name="context"><see cref="CodeBuilderContext"/> containing the necessary data to be written.</param>
+        /// <returns><see cref="CSharpCodeWriter"/> with the written content.</returns>
+        public static CSharpCodeWriter RenderWriteForTagHelperContent(
```

Comment:
This doesn't need to be here, move this into the `CSharpTagHelperCodeRenderer` as a private method (that way no need to pass writer/context..